core/src/main/java/org/apache/iceberg/jdbc/JdbcViewOperations.java

```diff
@@ -0,0 +1,190 @@
+import org.apache.iceberg.io.FileIO;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+import org.apache.iceberg.util.PropertyUtil;
+import org.apache.iceberg.view.BaseViewOperations;
+import org.apache.iceberg.view.ViewMetadata;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/** JDBC implementation of Iceberg ViewOperations. */
+public class JdbcViewOperations extends BaseViewOperations {
+
+  private static final Logger LOG = LoggerFactory.getLogger(JdbcViewOperations.class);
+  private final String catalogName;
+  private final TableIdentifier viewIdentifier;
+  private final FileIO fileIO;
+  private final JdbcClientPool connections;
+  private final Map<String, String> catalogProperties;
+
+  protected JdbcViewOperations(
+      JdbcClientPool dbConnPool,
+      FileIO fileIO,
+      String catalogName,
+      TableIdentifier viewIdentifier,
+      Map<String, String> catalogProperties) {
+    this.catalogName = catalogName;
+    this.viewIdentifier = viewIdentifier;
+    this.fileIO = fileIO;
+    this.connections = dbConnPool;
+    this.catalogProperties = catalogProperties;
+  }
+
+  @Override
+  protected void doRefresh() {
+    Map<String, String> view;
+
+    try {
+      view = JdbcUtil.getView(connections, catalogName, viewIdentifier);
+    } catch (InterruptedException e) {
+      Thread.currentThread().interrupt();
+      throw new UncheckedInterruptedException(e, "Interrupted during refresh");
+    } catch (SQLException e) {
+      // SQL exception happened when getting view from catalog
+      throw new UncheckedSQLException(
+          e, "Failed to get view %s from catalog %s", viewIdentifier, catalogName);
+    }
+
+    if (view.isEmpty()) {
+      if (currentMetadataLocation() != null) {
+        throw new NoSuchViewException("View does not exist: %s", viewIdentifier);
+      } else {
+        this.disableRefresh();
+        return;
+      }
+    }
+
+    String newMetadataLocation = view.get(JdbcTableOperations.METADATA_LOCATION_PROP);
+    Preconditions.checkState(
+        newMetadataLocation != null, "Invalid view %s: metadata location is null", viewIdentifier);
+    refreshFromMetadataLocation(newMetadataLocation);
+  }
+
+  @Override
+  protected void doCommit(ViewMetadata base, ViewMetadata metadata) {
+    boolean newView = base == null;
+    String newMetadataLocation = writeNewMetadataIfRequired(metadata);
+    try {
+      Map<String, String> view = JdbcUtil.getView(connections, catalogName, viewIdentifier);
+
+      if (base != null) {
+        validateMetadataLocation(view, base);
+        String oldMetadataLocation = base.metadataFileLocation();
+        // Start atomic update
+        LOG.debug("Committing existing view: {}", viewName());
+        updateView(newMetadataLocation, oldMetadataLocation);
+      } else {
+        // view not exists create it
+        LOG.debug("Committing new view: {}", viewName());
+        createView(newMetadataLocation);
+      }
+
```

Comment:
nit: redundant new line.  (please check all the places)

Thumb rule is that after every block (if, try, switch, for), we should have new line if it has some more statements. If no more statement, new line is not needed. 

Spotless don't have intelligence to fix this newline added as of now. 

Sorry for formatting nits. I hate these comments too :)  But good to keep the code in the same style as other files.

Reply:
OK, at least checkstyle can do a warning/error. That would be helpful. I will check if we have checkstyle/code formatter for IntelliJ. 
I will fix that. Thanks.